### PR TITLE
fix(argo-cd): Update CONTRIBUTING docs

### LINKED ISSUE #1781: `See Upgrade` link is broken
### Describe the bug

`See Upgrade` doesn't go anywhere.

### Related helm chart

other

### Helm chart version

N/A

### To Reproduce

1. Go to https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning
2. Click ([See Upgrade](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#upgrades))

### Expected behavior

Links to anchors should be to anchors that exist

### Screenshots

image

### Additional context

I'm trying to understand how to contribute to this repository.

Specifically, it appears that there are pairs of files, e.g. `charts/argo-cd/README.md` and `charts/argo-cd/README.md.gotmpl`. This file did not clarify that.

### PR DESCRIPTION
**Pull request by jmeridth**
Fixes #1781 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

Signed-off-by: jmeridth <jmeridth@gmail.com>
Co-authored-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]
LGTM

File: scripts/helm-docs.sh [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-echo "$CHART_DIR"
+REPO_ROOT="$(cd "$(dirname "$0")/.." && pwd)"
+echo "$REPO_ROOT"
 
 echo "Running Helm-Docs"
 docker run \
-    -v "$CHART_DIR:/helm-docs" \
+    -v "$REPO_ROOT:/helm-docs" \
     -u $(id -u) \
     jnorwood/helm-docs:v1.9.1
```
Comment (mbevc1) [inline scripts/helm-docs.sh:11]: Not a blocker, but is this worth bumping as well? If available, though.

Comment (jmeridth) [reply scripts/helm-docs.sh:11]: Yes. Good idea.

Comment (jmeridth) [reply scripts/helm-docs.sh:11]: https://hub.docker.com/layers/jnorwood/helm-docs/v1.11.0/images/sha256-e438eb9f879e4bb8389ce4acd0f7e9193e3d62e877a1a3cd5abd0393dbe55fe5?context=explore

1.11.0

**Review by pdrastil — APPROVED** [review]
LGTM

File: scripts/helm-docs.sh (hunk 1 above)
Comment (pdrastil) [reply scripts/helm-docs.sh:11]: If I remember right we haven't updated helm-docs as there is different behaviour for `##` comments that contain notes or related links. Newer versions outputs these values also into rendered README.md

Comment (mbevc1) [reply scripts/helm-docs.sh:11]: Happy to leave it as is if we're seeing issues.

Comment (jsoref) [reply scripts/helm-docs.sh:11]: The file isn't included in any charts and therefore doesn't appear to need to be reflected in any versioning.

The change here also doesn't change the behavior for charts going forward.

Comment (jmeridth) [reply scripts/helm-docs.sh]: @pdrastil I was seeing comments coming through on the current version (v1.9.1) in the file.  Upgrading gave same effect.  I just pushed the new version but can roll back.

Comment (jmeridth) [reply scripts/helm-docs.sh:11]: Created #1784 to be grabbed later

**Pull request merged**